src/Orleans/Utils/WrappedException.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace Orleans.Runtime.Internal
+{
+    /// <summary>Exception used to communicate with the storage provider, so that it throws this exception to its caller.</summary>
+    [Serializable]
+    public class WrappedException : OrleansException
```

Comment:
By doing it this way, we're adding to the Orleans public surface, imo, unnecessarily.
This behavior seems specific to the storage provider.
I don't see this as something that should be a general Orleans exception.